### PR TITLE
tests: Instrumentation Test for questionnaire fragment

### LINKED ISSUE #184: Add integration tests for the QuestionnaireFragment in the datacapture module
https://github.com/google/android-fhir/issues/184#issue-794091744

### PR DESCRIPTION
**Pull request by s-ayush2903**
Closes #184

### CODE REVIEW

File: datacapturegallery/src/androidTest/java/com/google/android/fhir/datacapture/gallery/QuestionnaireContainerFragmentTest.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  fun testQuestionnaireListElement3() {
+    clickRecyclerViewItems(R.string.questionnaire_list_entry_three_title)
+    onView(withText(R.string.questionnaire_list_entry_three_title)).check(matches(isDisplayed()))
+    onView(withText(R.string.submit_button_text)).check(matches(isDisplayed()))
+  }
+
+  @Test
+  fun testQuestionnaireListElement4() {
+    clickRecyclerViewItems(R.string.questionnaire_list_entry_four_title)
+    onView(withText(R.string.submit_button_text)).check(matches(isDisplayed()))
```
Comment (s-ayush2903) [inline datacapturegallery/src/androidTest/java/com/google/android/fhir/datacapture/gallery/QuestionnaireContainerFragmentTest.kt:75]: Needed to test via this text, as there are multiple occurrences of concerned questionnaire_title in the detailed questionnaire

**Comment by s-ayush2903** [comment]
Thoughts on mini DSL and Robot pattern? Last time you were unhappy with ui-automator, so now, only espresso is used here; just awaiting your idea on robot design. How does it look to you(needless?), I was unsure of your view, so kept it in a separate commit, so I won't be having any problem in deleting that commit.

File: datacapturegallery/src/androidTest/java/com/google/android/fhir/datacapture/gallery/QuestionnaireContainerFragmentTest.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    onView(withText(R.string.questionnaire_list_entry_four_text)).check(matches(isDisplayed()))
```
Comment (s-ayush2903) [inline datacapturegallery/src/androidTest/java/com/google/android/fhir/datacapture/gallery/QuestionnaireContainerFragmentTest.kt]: Had to test via specifically this text, as there were multiple occurrences of the title in the questionnaire

**Comment by s-ayush2903** [comment]
https://github.com/google/android-fhir/issues/184#issuecomment-805367964

**Comment by jingtang10** [comment]
I think the general approach is good. if you can apply this to the questionnaire fragment in the datacapture module that'd be great - thanks.

**Pull request closed without merging**